integrationTests/chainSimulator/bridge/executeOrDepositFail_test.go

```diff
@@ -371,3 +371,94 @@ func getTokenMetaData(token chainSim.ArgsDepositToken) *esdt.MetaData {
 		Creator: bytes.Repeat([]byte{1}, 32),
 	}
 }
+
+func TestChainSimulator_ExecuteSameNft2Times(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	whiteListedAddress := "erd1qqqqqqqqqqqqqpgqmzzm05jeav6d5qvna0q2pmcllelkz8xddz3syjszx5"
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   true,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              1,
+		GenesisTimestamp:         time.Now().Unix(),
+		RoundDurationInMillis:    uint64(6000),
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  0,
+		NumNodesWaitingListShard: 0,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.GeneralConfig.VirtualMachine.Execution.TransferAndExecuteByUserAddresses = []string{whiteListedAddress}
+			cfg.EpochConfig.EnableEpochs.DynamicESDTEnableEpoch = 0
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocksUntilEpochIsReached(3)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(0)
+
+	// Deploy bridge setup
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	argsEsdtSafe := ArgsEsdtSafe{
+		ChainPrefix:       sovChainPrefix,
+		IssuePaymentToken: "WEGLD-bd4d79",
+	}
+	initOwnerAndSysAccState(t, cs, initialAddress, argsEsdtSafe)
+	bridgeData := deployBridgeSetup(t, cs, initialAddress, esdtSafeWasmPath, argsEsdtSafe, feeMarketWasmPath)
+	chainSim.RequireAccountHasToken(t, cs, argsEsdtSafe.IssuePaymentToken, initialAddress, big.NewInt(0))
+
+	esdtSafeEncoded, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(bridgeData.ESDTSafeAddress)
+	require.Equal(t, whiteListedAddress, esdtSafeEncoded)
+
+	wallet, err := cs.GenerateAndMintWalletAddress(0, chainSim.InitialAmount)
+	require.Nil(t, err)
+	nonce := uint64(0)
+	paymentTokenAmount, _ := big.NewInt(0).SetString("1000000000000000000", 10)
+	chainSim.SetEsdtInWallet(t, cs, wallet, argsEsdtSafe.IssuePaymentToken, 0, esdt.ESDigitalToken{Value: paymentTokenAmount})
+
+	nftCollection := "NFTV2-a1b2c3"
+	bridgedInTokens := make([]chainSim.ArgsDepositToken, 0)
+	bridgedInTokens = append(bridgedInTokens, chainSim.ArgsDepositToken{
```

Comment:
Would've been a bit easier to read if you created a new var for depositToken and add the same pointer var two times